### ISSUE TITLE
Supports URL replacements like in markdown-link-check

### ISSUE THREAD

**Issue by boillodmanuel** [smallhadroncollider/brok#17]
From the [markdown-link-check](https://github.com/tcort/markdown-link-check#readme) readme:

> This behavior allows (for example) to adapt to certain platform conventions hosting the Markdown

In my example, it's for the gitlab wiki

**Comment by smallhadroncollider** [comment]
Yes, I can see that would be useful. Haven't used any of the Haskell RegEx packages yet, so will need to look into that side of things.

**Comment by boillodmanuel** [comment]
Here is an example:
- With a relative URL without extension  (`./info`), gitlab serves the (`./info.md`) page.
- With a relative URL with extension  (`./info.md`), gitlab serves the (`./info.md`) raw file.

Link to the gitlab doc that describes this behaviour: https://docs.gitlab.com/ee/user/markdown.html#wiki-specific-markdown

I also used the `ignorePatterns` capability of markdown-link-check.
